### PR TITLE
feat(excluded_filetypes): Add property to exclude filetypes in which …

### PR DESCRIPTION
**Pull request by brunnseb**
Add possibility to exclude certain filetypes from auto_suggestion being run. Manual suggestion still works as before.

### CODE REVIEW

File: lua/bropilot/util.lua [hunk 1]
```diff
@@ -64,6 +64,18 @@ local function finish_progress(handle)
   end
 end
 
+---@param tbl string[]
+---@param x string
+local function contains(tbl, x)
+  local found = false
+  for _, v in pairs(tbl) do
+    if v == x then
+      found = true
```
Comment (meeehdi-dev) [inline lua/bropilot/util.lua:73]: we could directly return true here instead of continuing the loop

File: lua/bropilot/util.lua [hunk 2]
```diff
@@ -64,6 +64,18 @@ local function finish_progress(handle)
...
+    end
+  end
+  return found
```
Comment (meeehdi-dev) [inline lua/bropilot/util.lua:76]: and if we end up returning true inside the loop, we can just ignore the `found` variable and return `false` here

**Review by meeehdi-dev — CHANGES_REQUESTED** [review]
hi @brunnseb ,

Thank you for your contribution 🙏 

The change looks good as is, but i think we can optimize the util function in case someone decides to have a huge array of excluded types.

**Comment by brunnseb** [comment]
> hi @brunnseb ,
> 
> Thank you for your contribution 🙏
> 
> The change looks good as is, but i think we can optimize the util function in case someone decides to have a huge array of excluded types.

Makes sense, I updated the `contains` helper accordingly

**Review by meeehdi-dev — APPROVED** [review]
Thank you 🫶

**Pull request merged**